warehouse/attestations/models.py

```diff
@@ -0,0 +1,48 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import typing
+
+from uuid import UUID
+
+from sqlalchemy import ForeignKey, orm
+from sqlalchemy.dialects.postgresql import CITEXT
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import Mapped, mapped_column
+
+from warehouse import db
+
+if typing.TYPE_CHECKING:
+    from warehouse.packaging.models import File
+
+
+class ReleaseFileAttestation(db.Model):
+    """
+    Association table between Release Files and Attestations.
```

Comment:
If we are storing the attestations on disk, do we need a separate table? Would a new field in the `File` table:
```python
attestation_paths: Mapped[list[str]] 
```
that stores the paths of the saved attestations work?

Reply:
I decided to store only the attestation digest and compute on the fly the `attestation_path` using the file digest.

Reply:
right, my question is rather: do we need this `ReleaseFileAttestation` association table? Since it's a one-to-many relationship, wouldn't a field in the `File` table be enough?

Reply:
The confusion came from the poorly named `ReleaseFileAttestation` and the wording used in the comment.

I've updated it :)